src/core/agent/agent.ts

```diff
@@ -328,6 +333,63 @@ class Agent {
     });
   }
 
+  async devPreload() {
+    try {
+      await SecureStorage.get(KeyStoreKeys.APP_PASSCODE);
+    } catch (error) {
+      if (
+        error instanceof Error &&
+        error.message ===
+          `${SecureStorage.KEY_NOT_FOUND} ${KeyStoreKeys.APP_PASSCODE}`
+      ) {
+        await SecureStorage.set(
+          KeyStoreKeys.APP_PASSCODE,
+          Agent.APP_PASSSCODE_DEV_MODE
+        );
+      }
+    }
+
+    try {
+      await SecureStorage.get(KeyStoreKeys.SIGNIFY_BRAN);
+    } catch (error) {
+      if (
+        error instanceof Error &&
+        error.message ===
+          `${SecureStorage.KEY_NOT_FOUND} ${KeyStoreKeys.SIGNIFY_BRAN}`
+      ) {
+        const branAndMnemonic = await Agent.agent.getBranAndMnemonic();
+        await SecureStorage.set(
+          KeyStoreKeys.SIGNIFY_BRAN,
+          branAndMnemonic?.bran
+        );
+      }
+    }
+    await PreferencesStorage.set(PreferencesKeys.APP_ALREADY_INIT, {
+      initialized: true,
+    });
+
+    await this.basicStorage.createOrUpdateBasicRecord(
+      new BasicRecord({
+        id: MiscRecordId.APP_PASSWORD_SKIPPED,
+        content: { value: true },
+      })
+    );
+
+    const keriaConnectUrlRecord = await this.basicStorage.findById(
+      MiscRecordId.KERIA_CONNECT_URL
+    );
+    const keriaBootUrlRecord = await this.basicStorage.findById(
+      MiscRecordId.KERIA_BOOT_URL
+    );
+
+    if (!keriaConnectUrlRecord && !keriaBootUrlRecord) {
+      await this.bootAndConnect({
+        url: ConfigurationService.env?.keri?.keria?.url || "",
+        bootUrl: ConfigurationService.env?.keri?.keria?.bootUrl || "",
+      });
```

Comment:
Hmm this isn't quite right.

Check if `KERIA_CONNECT_URL` and `KERIA_BOOT_URL` are already stored:
- If they are not stored, check if `ConfigurationService` values are not undefined
  - If undefined, return early and do nothing.
  - If not undefined, store them in the DB.
- Now use the stored values (that were either just stored now, or already existed) to call bootAndConnect if we didn't return early

Reply:
Because `bootAndConnect` has a fragment called saveAgentUrls to store the ConfigurationService values. I think we don't need to store them before calling `bootAndConnect`

Reply:
Good point! We can skip that. But you still need to cover the other scenarios:
- If the URLs already exist, we need to just boot using them directly and not call `bootAndConnect`
- If they don't exist, and the `ConfigurationService` values are empty, we should not call `bootAndConnect`

Reply:
For point 1, I think if the URLs already exists, it means that signify is already boot with `bran`. So if re-boot with that `bran`, i think it will not work

Reply:
That's a good point. But in that case we should never be calling `bootAndConnect` or calling `setOnlineStatus`.

Can we simply do this:
- For each URL check if it does not exist already
- If it doesn't, and the value in `ConfigurationService` is not empty, store the record in the DB (do not boot)
- Continue on and the rest of `initApp` will boot it with these URLs if they exist

Reply:
resolving - we decided to not change anything here because sometimes we need to boot and other times we don't. Easier to let the user boot